sled-agent/src/bootstrap/server.rs

```diff
@@ -26,7 +26,7 @@ impl Server {
         sled_config: SledConfig,
     ) -> Result<Self, String> {
         let (drain, registration) = slog_dtrace::with_drain(
-            config.log.to_logger("bootstrap-agent").map_err(|message| {
+            config.log.to_logger("SledAgent").map_err(|message| {
```

Comment:
To make sure I understand, the bootstrap server sets the key `"name" => "SledAgent"`, and then the remaining items set the `"component"` key? It looks like the sled agent server itself never sets a component key. That could be fine, I'm just checking that's the intended behavior. While it might be duplicative, it could also be nice to give it a component anyway, so that we can filter the log messages explicitly from the sled agent server by component name as opposed to finding the messages _without_ a component key. Thoughts?

Reply:
> To make sure I understand, the bootstrap server sets the key `"name" => "SledAgent"`, and then the remaining items set the `"component"` key?

That's right - I looked at the `bunyan` source code, and it seems to emit a "Name string" of "name" + "component" + "pid":

https://github.com/trentm/node-bunyan/blob/0ff1ae29cc9e028c6c11cd6b60e3b90217b66a10/bin/bunyan#L820-L829

With this new formatting, some example logs, piped to the bunyan tool:

```
[2022-05-12T18:17:18.931110745Z]  INFO: SledAgent/StorageManager/2735 on thelio: Initialiting Clickhouse (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                                       
[2022-05-12T18:17:19.471949566Z]  INFO: SledAgent/StorageManager/2735 on thelio: Zone oxz_clickhouse_oxp_d462a7f7-b628-40fe-80ff-4e4189e2d62b with address [fd00:1122:3344:101::5]:8123 is running (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                                                                                                                                                                                           
[2022-05-12T18:17:19.472912767Z]  INFO: SledAgent/dropshot/2735 on thelio: request completed (req_id=b5a0e0c4-c479-443e-b56c-00003aba7637, uri=/filesystem, method=PUT, remote_addr=[fd00:1122:3344:101::1]:62657, local_addr=[fd00:1122:3344:101::1]:12345, response_code=204)                                                                                                                                                                                                             [2022-05-12T18:17:19.474461Z]  INFO: SledAgent/RSS/2735 on thelio: Finished setting up agents and datasets                                                                                                                                    
[2022-05-12T18:17:19.480097997Z]  INFO: SledAgent/RSS/2735 on thelio: sending service requests...                      
[2022-05-12T18:17:19.480890282Z]  INFO: SledAgent/RSS/2735 on thelio: initializing sled services: [ServiceRequest { name: "nexus", addresses: [fd00:1122:3344:101::3], gz_addresses: [] }, ServiceRequest { name: "oximeter", addresses: [fd00:1122:3344:101::4], gz_addresses: [] }, ServiceRequest { name: "internal-dns", addresses: [fd00:1122:3344:1::1], gz_addresses: [fd00:1122:3344:1::2] }]                                                                                       
[2022-05-12T18:17:19.482956144Z]  INFO: SledAgent/dropshot/2735 on thelio: accepted connection (local_addr=[fd00:1122:3344:101::1]:12345, remote_addr=[fd00:1122:3344:101::1]:52938)                                                          
[2022-05-12T18:17:19.483727915Z]  INFO: SledAgent/ServiceManager/2735 on thelio: Ensuring services are initialized: [ServiceRequest { name: "nexus", addresses: [fd00:1122:3344:101::3], gz_addresses: [] }, ServiceRequest { name: "oximeter", addresses: [fd00:1122:3344:101::4], gz_addresses: [] }] (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                                                                                      
[2022-05-12T18:17:19.484264417Z]  INFO: SledAgent/ServiceManager/2735 on thelio: Service nexus does not yet exist (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                              
[2022-05-12T18:17:19.495517967Z]  INFO: SledAgent/ServiceManager/2735 on thelio: Configuring new Omicron zone: oxz_nexus (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                       
[2022-05-12T18:17:19.512471396Z]  INFO: SledAgent/ServiceManager/2735 on thelio: Installing Omicron zone: oxz_nexus (sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                            [2022-05-12T18:17:23.438319233Z]  INFO: SledAgent/ServiceManager/2735 on thelio: Zone booting (zone=oxz_nexus, sled_id=fb0f7546-4d46-40ca-9d56-cbb810684ca7)                                                                                  [2022-05-12T18:17:25.102185737Z]  WARN: SledAgent/2735 on thelio: failed to contact nexus, will retry in 2.910554634s                                                                                                                         
```

> It looks like the sled agent server itself never sets a component key. That could be fine, I'm just checking that's the intended behavior. While it might be duplicative, it could also be nice to give it a component anyway, so that we can filter the log messages explicitly from the sled agent server by component name as opposed to finding the messages _without_ a component key. Thoughts?

Sure, I can adjust this. It'll require passing a logger to the sled agent constructor which *does not* have `component` set, and setting it internally.

Reply:
In the existing implementation, what do lines from the sled agent itself show? `SledAgent/<PID>`, I think? I don't feel super strongly, I just want to make sure we can do things like select just the lines from the main sled agent (man, we need a better name!), just the service manager, etc.

Reply:
In [1552ac0](https://github.com/oxidecomputer/omicron/pull/1060/commits/1552ac0013952a1ecc6ec871917dfa9b3d1bfa59), it was `SledAgent/<PID>`. On the more recent [2d06891](https://github.com/oxidecomputer/omicron/pull/1060/commits/2d068917da477a5b3c324f729bb16fb36d6cce14), it'll now be `SledAgent/SledAgent/<PID>` - but most of the sled agent work goes to subcomponents, so this basically only appears during setup.

Reply:
Makes sense, the choice is yours :) The fact that most of the work is done by other components is helpful.